From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Wei Fu <wefu@redhat.com>
Date: Sat, 2 Apr 2022 22:53:36 +0800
Subject: [PATCH] Fix the -march issue for riscv64

There is an issue on riscv64 system when compiling it natively:
gcc: error: '-march=native': ISA string must begin with rv32 or rv64

This patch set HOST_MARCH= like ia64 to resolve the issue.

Signed-off-by: Wei Fu <wefu@redhat.com>
(cherry picked from commit aab4e9b10ac9e98588a1b19771cf6f4c8c0a3096)
---
 src/include/defaults.mk | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/include/defaults.mk b/src/include/defaults.mk
index 632b155..b8cc590 100644
--- a/src/include/defaults.mk
+++ b/src/include/defaults.mk
@@ -73,10 +73,14 @@ override SOFLAGS = $(_SOFLAGS) \
 
 HOST_ARCH=$(shell uname -m)
 ifneq ($(HOST_ARCH),ia64)
+ifneq ($(HOST_ARCH),riscv64)
 	HOST_MARCH=-march=native
 else
 	HOST_MARCH=
 endif
+else
+	HOST_MARCH=
+endif
 HOST_CPPFLAGS ?= $(CPPFLAGS)
 override _HOST_CPPFLAGS := $(HOST_CPPFLAGS)
 override HOST_CPPFLAGS = $(_HOST_CPPFLAGS) \
